src/Training/Bundle/UserNamingBundle/Resources/config/services.yml

```diff
@@ -0,0 +1,23 @@
+services:
+
+  training_user_name_formatter_container:
+    class: Training\Bundle\UserNamingBundle\Container\UserNamePartsContainer
+
+  training_user_name_formatter:
+    class: Training\Bundle\UserNamingBundle\Formatter\UserNameFormatter
+    arguments:
+      - '@training_user_name_formatter_container'
+
+  oro_locale.entity_name_provider.composite:
```

Comment:
We usually name decorators - decorators. [Composite](https://refactoring.guru/design-patterns/composite) DP is totally unrelated to this class